android/src/main/java/com/swmansion/rnscreens/ScreenStackFragment.kt

```diff
@@ -568,4 +462,14 @@ class ScreenStackFragment :
         }
         return sheetDelegate!!
     }
+
+    private fun requireSheetAnimationDelegate(
+        sheetDelegate: SheetDelegate,
+        dimmingDelegate: DimmingViewManager,
+    ): SheetAnimationDelegate {
+        if (sheetAnimationDelegate == null) {
+            sheetAnimationDelegate = SheetAnimationDelegate(this, screen, coordinatorLayout, sheetDelegate, dimmingDelegate)
+        }
+        return sheetAnimationDelegate!!
+    }
 }
```

Comment:
Don't take & store `sheetDelegate` & `dimmingDelegate` inside `SheetAnimationDelegate`. This only potentially creates retain cycles. 
 
Instead pass the information needed as arguments to particular function invocations.

Reply:
https://github.com/software-mansion/react-native-screens/pull/3378/commits/7f42ff38544f1d3d43851e4b43c776d4a9ab070b